projects/laji/src/app/shared-modules/document-viewer/images/images.component.ts

```diff
@@ -50,13 +51,31 @@ export class ImagesComponent implements OnChanges {
           this.gatheringImages = this.gatheringImages.concat(gathering.media);
         }
         if (gathering.units) {
-          gathering.units.filter(unit => unit.unitId === this.highlight).map(
+          if (!this.identifying) {
+            gathering.units.filter(unit => unit.unitId === this.highlight).map(
               item => {
                 if (item.media) {
                   this.unitImages = this.unitImages.concat(item.media);
                 }
               }
-          );
+            );
+          } else {
+            let imgUnitId;
+            gathering.units.forEach(element => {
+              (element.media || []).forEach(image => {
+                if (image.fullURL === this.highlight) {
+                  imgUnitId = element.unitId;
+                }
+              });
+            });
```

Comment:
Unnecessary `forEach` iteration, would be more performant with `find` (something like `const imgUnitId = gatherings.units.find(...`). The same goes for the inner `forEach`